File: ru/javastudy/main/homework5/Box.java
package ru.javastudy.homework5;

public record Box<T, U, V>(T first, U second, V third) {

    public static void main(String[] args) {
        BoxTest.testBoxes();
    }

    public float getSum() {
        float sum = 0.0F;

        if (first instanceof Number) sum += ((Number) first).floatValue();
        if (second instanceof Number) sum += ((Number) second).floatValue();
        if (third instanceof Number) sum += ((Number) third).floatValue();

        return sum;
    }

    public static class BoxTest {
        public static void testBoxes() {
            Box<String, Float, Integer> firstBox = new Box<>("test", 5.5F, 30);
            Box<Double, Integer, String> secondBox = new Box<>(15.9D, 127, "test 2");
            Box<Long, String, Integer> thirdBox = new Box<>(99L, "test 3", 15);

            float totalSum = firstBox.getSum() + secondBox.getSum() + thirdBox.getSum();
            System.out.println("\nСумма всех переданных в коробки чисел: " + totalSum);
        }
    }
}

File: main/homework5/Box.kt
package ru.kotlinstudy.main.homework5


class Box<T, U, V>(
    private val first: T,
    private val second: U,
    private val third: V
) {
    fun getSum(): Double {
        var sum = 0.0

        if (first is Number) sum += first.toFloat()
        if (second is Number) sum += second.toFloat()
        if (third is Number) sum += third.toFloat()

        return sum
    }
}

fun main() {
    val firstBox = Box("text", 10, 5.5F)
    val secondBox = Box(20, 3.14, "text 2")
    val thirdBox = Box(7.5F, "text 3", 15)

    val totalSum = firstBox.getSum() + secondBox.getSum() + thirdBox.getSum()
    println("\nСумма всех переданных в коробки чисел: $totalSum")
}